cwms-data-api/src/main/java/cwms/cda/data/dto/locationlevel/VirtualLocationLevel.java

```diff
@@ -0,0 +1,352 @@
+			this.attributeComment = attributeComment;
+			return this;
+		}
+
+		@Override
+		public VirtualLocationLevel.Builder withLocationLevelId(String locationId) {
+			this.locationId = locationId;
+			return this;
+		}
+
+		@Override
+		public VirtualLocationLevel.Builder withOfficeId(String officeId) {
+			this.officeId = officeId;
+			return this;
+		}
+
+		@Override
+		public VirtualLocationLevel.Builder withInterpolateString(String interpolateString) {
+			this.interpolateString = interpolateString;
+			return this;
+		}
+
+		@Override
+		public VirtualLocationLevel build()
+		{
+			return new VirtualLocationLevel(this);
+		}
+	}
+
+	@JsonPOJOBuilder
+	@JsonNaming(PropertyNamingStrategies.KebabCaseStrategy.class)
+	@JsonDeserialize(builder = Constituent.Builder.class)
+	@FormattableWith(contentType = Formats.JSONV2, formatter = JsonV2.class, aliases = {Formats.DEFAULT, Formats.JSON})
+	public static final class Constituent
+	{
+		private final String abbr;
+		private final String type;
+		private final String name;
+		private final String attributeId;
+		private final Double attributeValue;
+		private final String attributeUnits;
+
+		private Constituent(Builder builder)
+		{
+			this.abbr = builder.abbr;
+			this.type = builder.type;
+			this.name = builder.name;
+			this.attributeId = builder.attributeId;
+			this.attributeValue = builder.attributeValue;
+			this.attributeUnits = builder.attributeUnits;
+		}
+
+		public String getAbbr()
+		{
+			return abbr;
+		}
+
+		public String getType()
+		{
+			return type;
+		}
+
+		public String getName()
+		{
+			return name;
+		}
+
+		public String getAttributeId()
+		{
+			return attributeId;
+		}
+
+		public Double getAttributeValue()
+		{
+			return attributeValue;
+		}
+
+		public String getAttributeUnits()
+		{
+			return attributeUnits;
+		}
+
+		public List<String> getConstituentList()
+		{
+			List<String> retVal = new ArrayList<>();
+			retVal.add(abbr);
+			retVal.add(type);
+			retVal.add(name);
+			if(attributeId != null)
+			{
+				retVal.add(attributeId);
+			}
+			if(attributeValue != null)
+			{
+				retVal.add(attributeValue.toString());
+			}
+			if(attributeUnits != null)
+			{
+				retVal.add(attributeUnits);
+			}
+			return retVal;
+		}
+
+		@JsonPOJOBuilder
+		@JsonNaming(PropertyNamingStrategies.KebabCaseStrategy.class)
+		@JsonInclude(JsonInclude.Include.NON_NULL)
+		public static final class Builder
+		{
+			private final String abbr;
+			private final String type;
+			private final String name;
+			private String attributeId;
+			private Double attributeValue;
+			private String attributeUnits;
+
+			@JsonCreator
+			public Builder(@JsonProperty(value = "abbr", required = true) String abbr,
+					@JsonProperty(value = "type", required = true) String type,
+					@JsonProperty(value = "name", required = true) String name)
+			{
+				this.abbr = abbr;
+				this.type = type;
+				this.name = name;
+			}
+
+			public Builder withAttributeId(String constituentAttributeId)
```

Comment:
Attributes are common to all levels, and this is missing all the other common data